pkg/apis/pingcap/v1alpha1/tidbcluster.go

```diff
@@ -300,7 +300,11 @@ func (tc *TidbCluster) TiDBAllMembersReady() bool {
 }
 
 func (tc *TidbCluster) TiDBStsDesiredReplicas() int32 {
-	return tc.Spec.TiDB.Replicas + int32(len(tc.Status.TiDB.FailureMembers))
+	if tc.Spec.TiDB.Replicas == int32(0) {
+		return tc.Spec.TiDB.Replicas
+	} else {
+		return tc.Spec.TiDB.Replicas + int32(len(tc.Status.TiDB.FailureMembers))
+	}
```

Comment:
I tend not to change this method to keep its semantic simple, how to handle `failureMembers` when replias is 0 is just another concern.

Reply:
If i don't change this method, I will need to change this code 	https://github.com/pingcap/tidb-operator/blob/5ead47e9d230ed9bf0a6acd2d2a9cbde729e6c48/pkg/manager/member/tidb_member_manager.go#L756 replicas and all methods using `TiKVStsDesiredReplicas` method .
I think change this method easier to understand. When scaling to 0 replicas,the  desired replicas is 0. Right?

Reply:
Yes, but **that is exactly** the sematic I suppose, we do not magically ignore `failureMembers` in `tidbcluster` when `.spec.replica` is some specific value (e.g. 0), instead, we let the owner of `failureMembers`, aka `TiDBFailover` to manage the failureMembers.

Reply:
I get.